src/steps/10-FinancialDetails/IncrementalFunding.spec.ts

```diff
@@ -0,0 +1,628 @@
+      wrapper.vm.initializeIncrements();
+
+      expect(wrapper.vm.$data.fiscalQuarters.length).toBe(6);
+    });
+
+    it("hasChanges() - setting different currentData and savedData object" +
+        "to set hasChanges to true", async () => {
+      wrapper.setData({
+        currentData: {
+          initialFundingIncrementStr: "$1.00",
+          fundingIncrements: [
+            {
+              text: "text01",
+              amt: "1.01",
+              order: 1,
+              sysId: "id_01",
+              qtrOrder: 1, 
+              hasPeriodGap: false,            
+            }
+          ]
+        },
+        savedData: {
+          initialFundingIncrementStr: "$2.00",
+          fundingIncrements: [
+            {
+              text: "text01",
+              amt: "1.01",
+              order: 1,
+              sysId: "id_01",
+              qtrOrder: 1, 
+              hasPeriodGap: false,            
+            }
+          ]
+        }
+      });
+      const _hasChanged = wrapper.vm.hasChanged();
+      expect(_hasChanged).toBe(true);
+    });
+
+    it("hasChanges() - setting same currentData and savedData object" +
+        "to set hasChanges() to false", async () => {
+      wrapper.setData({
+        currentData: {
+          initialFundingIncrementStr: "1.00",
+          fundingIncrements: []
+        },
+        savedData: {
+          initialFundingIncrementStr: "1.00",
+          fundingIncrements: []
+        }
+      });
+      const _hasChanged = wrapper.vm.hasChanged();
+      expect(_hasChanged).toBe(false);
+    });
+
+
+    it("quarterChange() - supplies correct args/data to successfully change  " +
+        " data.fundingIncrements[]", async () => {
+      wrapper.setData({
+        fundingIncrements: fundingIncrements,
+        fiscalQuarters: fiscalQuarters,
+        selectedQuarters: selectedQuarters
+      })
+      const _fundingIncs = wrapper.vm.$data.fundingIncrements;
+      expect(_fundingIncs[0].text).toEqual("4th QTR FY22");
+      expect(_fundingIncs[1].text).toEqual("1st QTR FY23");
+      const args = {
+        newSelectedValue: {
+          multiSelectOrder: 4,
+          text: "3rd QTR FY23",
+        },
+        selectedBeforeChange: {
+          multiSelectOrder: 1,
+          text: "4th QTR FY22",
+        }
+      }
+
+      wrapper.vm.quarterChange(args);
+      expect(_fundingIncs[0].text).toEqual("1st QTR FY23");
+      expect(_fundingIncs[1].text).toEqual("3rd QTR FY23");
+    });
+  });
+
+
+  it("validateOnContinue() - sets valid outOfRangeIndex with selectedQuarters " +
+      "so that data.allowContinue will set to false", async () => {
+    await wrapper.setData({
+      hasValidatedOnContinue: false,
+      outOfRangeIndex: 1,
+      selectedQuarters: [
+        { multiSelectOrder: 1, text: "4th QTR FY22"},
+        { multiSelectOrder: 2, text: "2nd QTR FY23"},
+      ]
+    })
+    await wrapper.vm.validateOnContinue();
+    expect(wrapper.vm.$data.allowContinue).toBe(false);
```

Comment:
Thx, but this route was chosen because the data properties reactivity in Vue is adequate for this
```    
if (!this.hasValidatedOnContinue && (this.outOfRangeIndex && this.outOfRangeIndex >= 0 
      || this.isIFPUnderfunded || this.isIFPOverfunded)
    ) {
      this.allowContinue = false;
    } else {
      this.allowContinue = true;
    }
```